code/parachain/runtime/dali/src/lib.rs

```diff
@@ -1085,21 +1086,29 @@ parameter_types! {
 }
 
 /// Native <-> Cosmwasm account mapping
-/// TODO(hussein-aitlahcen): Probably nicer to have SS58 representation here.
 pub struct AccountToAddr;
 impl Convert<alloc::string::String, Result<AccountId, ()>> for AccountToAddr {
 	fn convert(a: alloc::string::String) -> Result<AccountId, ()> {
-		match a.strip_prefix("0x") {
-			Some(account_id) => Ok(<[u8; 32]>::try_from(hex::decode(account_id).map_err(|_| ())?)
-				.map_err(|_| ())?
-				.into()),
-			_ => Err(()),
+		if let Some(hex_account) = a.strip_prefix("0x") {
```

Comment:
Do we really want to allow both hex and SS58?

Reply:
Actually, I don't see a use case where users/developers want to use hex over SS58. I provided that feature just in case but let me remove it and if we ever see a need or use case, then we add.

Reply:
Great!